<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>397</width>
    <height>297</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Pynguin Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Files</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Saving</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QRadioButton" name="savesingle">
            <property name="text">
             <string>Single file (*.pyn)</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="savefolder">
            <property name="text">
             <string>Directory of files (*.pyn + *_pynd)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>External files (*.py)</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QCheckBox" name="reloadexternal">
            <property name="text">
             <string>Reload changed file automatically</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="autorun">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Auto Test/Run after reloading</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Backups</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="maximumSize">
            <size>
             <width>80</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Keep</string>
           </property>
           <property name="margin">
            <number>15</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="backupkeep">
           <property name="maximumSize">
            <size>
             <width>50</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>backup files</string>
           </property>
           <property name="margin">
            <number>15</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="maximumSize">
            <size>
             <width>80</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Rate</string>
           </property>
           <property name="margin">
            <number>16</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="backuprate">
           <property name="maximumSize">
            <size>
             <width>50</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>60</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>minutes between backups</string>
           </property>
           <property name="margin">
            <number>15</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Location</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Folder</string>
              </property>
              <property name="margin">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="backupfolderpath"/>
            </item>
            <item>
             <widget class="QPushButton" name="backupbrowseButton">
              <property name="text">
               <string>Browse</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>File name</string>
              </property>
              <property name="margin">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="backupfilename">
              <property name="text">
               <string>backup~%s.pyn</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>(must contain %s)</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Editor</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Main function / class</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QRadioButton" name="editor_mainfirst">
            <property name="text">
             <string>First function/ class on the page</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="editor_mainlast">
            <property name="text">
             <string>Last function/ class on the page</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Test / Run all settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QCheckBox" name="testrun_reset">
            <property name="text">
             <string>reset() between pages</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="testall_reverse">
            <property name="text">
             <string>&quot;Test/ Run all&quot; in reverse order (last to first)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="testall_autocall">
            <property name="text">
             <string>Automatically call main function/ class</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Console</string>
      </attribute>
      <widget class="QCheckBox" name="quietinterrupt">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>50</y>
         <width>173</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Quiet KeyboardInterrupt</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>Colors</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_11">
       <item>
        <widget class="QLabel" name="label_13">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Startup Colors</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Background color</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="bgcolor">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Pen color</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="pencolor">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Fill color</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="fillcolor">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>38</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="set_colors_to_current">
         <property name="text">
          <string>Set to Current Colors</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>37</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>Other</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_10">
       <item>
        <widget class="QCheckBox" name="reset_forces_visible">
         <property name="text">
          <string>reset() forces visible pynguin</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="allow_start_hidden">
         <property name="text">
          <string>Allow starting with pynguin hidden</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>139</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Reset all settings to defaults</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="defaultsettings">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>* Warning: Also resets drawing!</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backupbrowseButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>backupbrowse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reloadexternal</sender>
   <signal>stateChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>externaloption()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backupkeep</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>backupoption()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>defaultsettings</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>defaultsettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bgcolor</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>bgcolorbrowse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pencolor</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>pencolorbrowse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fillcolor</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>fillcolorbrowse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>124</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>set_colors_to_current</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>set_colors_to_current()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>backupbrowse()</slot>
  <slot>externaloption()</slot>
  <slot>backupoption()</slot>
  <slot>defaultsettings()</slot>
  <slot>bgcolorbrowse()</slot>
  <slot>pencolorbrowse()</slot>
  <slot>fillcolorbrowse()</slot>
  <slot>set_colors_to_current()</slot>
 </slots>
</ui>
